core/server/master/src/main/java/alluxio/master/block/meta/WorkerMeta.java

```diff
@@ -0,0 +1,34 @@
+package alluxio.master.block.meta;
+
+import alluxio.StorageTierAssoc;
+import alluxio.util.CommonUtils;
+import alluxio.wire.WorkerNetAddress;
+import com.google.common.base.Preconditions;
+
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class WorkerMeta {
+  /** Worker's address. */
+  public final WorkerNetAddress mWorkerAddress;
+  /** The id of the worker. */
+  public final long mId;
+  /** Start time of the worker in ms. */
+  public final long mStartTimeMs;
+  /** Worker's last updated time in ms. */
+  public final AtomicLong mLastUpdatedTimeMs;
```

Comment:
I'm using Atomic here because I want to make it easier to read and update this field without locking. The RW lock on WorkerMeta is actually just for the `mIsRegistered` field. It might be harder to get rid of that lock totally, because when we write lock the `WorkerMeta` we want the `mIsRegistered` field to be mutex. But honestly the only corner case that can lead to this race condition is two parallel requests registering the same worker. I'm not sure if that should be handled (now the behavior is not too well defined).

Reply:
it looks to me this is the only mutable part. How about making this meta info constant, like `StaticWorkerMeta` or even putting them back to `MasterWorkerInfo`, and moving `mLastUpdatedTimeMs` out to other meta part? In this way, accessing these readonly fields requires no locks, eliminating one RW lock.

Reply:
I managed to make this atomic so the locking on this field is unnecessary. The `mIsRegistered` field has a check-then-act section so still the lock is needed. I moved `mIsRegistered` to the upper level `MasterWorkerInfo` and use an exclusive lock for that now. Now the `WorkerMeta` class is thread safe (not static because of the last updated ts).

Reply:
Moved `mLastUpdatedTimsMs` out of `WorkerMeta` to `MasterWorkerInfo`